user creation and removal

rlebre opened the pull request:
Added section in Configuration page about creation and removal of users. However, this pull request only takes into account the [dev](https://github.com/bioinformatics-ua/dicoogle/tree/dev) branch in [bioinformatics-ua/dicoogle](https://github.com/bioinformatics-ua/dicoogle/tree/dev) as it considers the refactored user web services.

Solves #11

Enet4 commented on _docs/configuring.md:
````diff
@@ -27,15 +27,15 @@ In the Management menu, Services and Plugins tab, it is possible to start and/or
+All configurations previously mentioned are stored in a single XML file. Once Dicoogle is run at least once, you will find a file named _"config.xml"_. Currently, some configurations can only be changed by editing this file. As an example, let us modify this instance's application entity title (AE Title). Look for the XML element `AETitle` :
 
-```xml
+``` xml
````
This is not necessary, and I doubt it's idiomatic. Did any editor software complain about this?

```suggestion
```xml
```

Enet4 commented on _docs/configuring.md:
````diff
@@ -27,15 +27,15 @@ In the Management menu, Services and Plugins tab, it is possible to start and/or
 And modify it to another identifier:
 
-```xml
+``` xml
````
Same.

rlebre replied:
solved in https://github.com/bioinformatics-ua/dicoogle-learning-pack/pull/13/commits/1ed27d9b2d294d0a73b645dbba0a14173947dd70

rlebre commented:
@Enet4 in fact, I think v2.5.0 only supports the HTTP method PUT to create a user. Changes have been made to change the API from PUT to POST and change also the DELETE method. The pull request referring that is https://github.com/bioinformatics-ua/dicoogle/pull/380

Regarding the white spaces commented above, they were added by markdown-formatter plugin for VSCode. Although there was not complain by an editor, I can revert upon your confirmation about this matter.

Enet4 commented:
Right, now I remember that change. You can document it as `PUT` in the main corpus and append an "unreleased" block to document the next version.

```html
<div class="note unreleased" >
<h5>Breaking change in Dicoogle 3</h5>
<p>Starting from Dicoogle 3.0.0, creating new users should be done with the `POST` method instead of `PUT`.</p>
</div>
```

rlebre commented:
Also fixed the wrong method to delete users.
Moreover, I added the information of the persistence of the users in users.xml to the note unreleased, as the fixes are applied only in v3.0.0 by https://github.com/bioinformatics-ua/dicoogle/pull/380

Enet4 commented on _docs/configuring.md:
````diff
@@ -64,6 +64,41 @@ And change it to our storage:
+To create a new user in the system, one can perform a HTTP PUT in `/user` with `username` and `password` query strings. Example:
+
+``` bash
+  curl -X PUT "http://localhost:8080/user?username=johndoe&password=secret"
````
Indentation inside code fence blocks isn't required. I think it will look good enough without it.

```suggestion
curl -X PUT "http://localhost:8080/user?username=johndoe&password=secret"
```

Enet4 commented on _docs/configuring.md:
````diff
@@ -64,6 +64,41 @@ And change it to our storage:
+  curl -X PUT "http://localhost:8080/user?username=johndoe&password=secret"
+```
+
+Alternatively, one may create an administrator user, adding the flag `admin` set to `true` :
````
```suggestion
Alternatively, one may create an administrator user by adding the flag `admin` set to `true`:
```

Enet4 commented on _docs/configuring.md:
````diff
@@ -64,6 +64,41 @@ And change it to our storage:
+Alternatively, one may create an administrator user, adding the flag `admin` set to `true` :
+
+``` bash
+  curl -X PUT "http://localhost:8080/user?username=johndoe&password=secret&admin=true"
````
```suggestion
curl -X PUT "http://localhost:8080/user?username=johndoe&password=secret&admin=true"
```

Enet4 commented on _docs/configuring.md:
````diff
@@ -64,6 +64,41 @@ And change it to our storage:
+The removal of the user is executed with the HTTP method DELETE and the parameters are passed as path query to the same endpoint ( `/users` ). Example:
+
+``` bash
+  curl -X DELETE "http://localhost:8080/user/johndoe"
````
```suggestion
curl -X DELETE "http://localhost:8080/user/johndoe"
```

Enet4 commented on _docs/configuring.md:
```diff
@@ -64,6 +64,41 @@ And change it to our storage:
+  <h5>Breaking changes in Dicoogle 3</h5>
+  <ul>
+    <li>Starting from Dicoogle 3.0.0, creating new users should be done with the POST method instead of PUT.</li>
+    <li>Also from Dicoogle 3.0.0 on, bugs were fixed and the actions performed in these endpoints are reflected in the _users.xml_ configuration file and, thus, persisted.</li>
```
We might still wish to throw in the fix for Dicoogle 2 eventually, since we don't want to drop support for v2 completely, right after v3 is released. For the time being, we can either state that future versions will persist users properly or just leave it like this.

rlebre replied:
I think you are right, I just crafted that part of the text and turned it more generic

Enet4 reviewed (approved):
Let's bring in pending changes and work on fresh new content for the learning pack!
